velox/exec/HashProbe.h

```diff
@@ -55,7 +87,26 @@ class HashProbe : public Operator {
   void clearDynamicFilters() override;
 
  private:
-  // Sets up 'filter_' and related members.
+  void setState(State state);
+  void stateTransitionCheck(State state);
+
+  void setRunning();
+  void checkRunning() const;
+  bool isRunning() const;
+
+  void waitForTableBuild();
+
+  // Invoked to set up spilling related input processing. The function sets up a
+  // reader to read probe inputs from spilled data on disk if
+  // 'restoredSpillPartitionId' is not null. If 'spillPartitionIds' is not
+  // empty, then spilling has been triggered at the build side and the function
+  // will set up a spiller and the associated data structures to spill probe
+  // inputs.
+  void maybeSetupSpillInput(
```

Comment:
I assume that only one of the parameters can be non-empty, e.g. it is not possible both restoredSpillPartitionId and spillPartitionIds to be non-empty. Is this so? Would be nice to clarify in the comments and perhaps check in the implementation.

Reply:
Got it. Thank you for explaining. Curious if we have test coverage for recursive spilling.